docs/source/format/ADBC.rst

```diff
@@ -0,0 +1,307 @@
+  so on. Provides a single client in C++ and Java language that talks
+  to any database servers implementing the protocol. Models its API
+  surface (though not API design) after JDBC and ODBC.
+
+`DBI for R <https://www.r-dbi.org/>`_
+  An R package/ecosystem of packages for database access. Provides a
+  single interface with “backends” for specific databases.  While
+  row-oriented, `integration with Arrow is under consideration`_,
+  including a sketch of effectively the same idea as ADBC.
+
+`JDBC <https://jcp.org/en/jsr/detail?id=221>`_
+  A Java library for database access, providing row-based
+  APIs. Provides a single interface with drivers for specific
+  databases.
+
+`ODBC <https://github.com/microsoft/ODBC-Specification>`_
+  A language-agnostic standard from the ISO/IEC for database access,
+  associated with Microsoft. Feature-wise, it is similar to JDBC (and
+  indeed JDBC can wrap ODBC drivers), but it offers columnar data
+  support through fetching buffers of column values. (See above for
+  caveats.) Provides a single C interface with drivers for specific
+  databases.
+
+`PEP 249 <https://www.python.org/dev/peps/pep-0249/>`_ (DBAPI 2.0)
+  A Python standard for database access providing row-based APIs. Not
+  a singular package, but rather a set of interfaces that packages
+  implement.
+
+Existing libraries
+------------------
+
+These are libraries which either 1) implement columnar data access for
+a particular system; or 2) could be used to implement such access.
+
+:doc:`Arrow Flight <./Flight>`
+  An RPC framework optimized for transferring Arrow record batches,
+  with application-specific extension points but without any higher
+  level semantics.
+
+:doc:`Arrow JDBC <../java/jdbc>`
+  A Java submodule, part of Arrow/Java, that uses the JDBC API to
+  produce Arrow data. Internally, it can read data only row-at-a-time.
+
+`arrow-odbc <https://github.com/pacman82/arrow-odbc>`_
+  A Rust community project that uses the ODBC API to produce Arrow
+  data, using ODBC’s buffer-based API to perform bulk copies. (See
+  also: Turbodbc.)
+
+`Arrowdantic <https://github.com/jorgecarleitao/arrowdantic/>`_
+  Python bindings for an implementation of ODBC<>Arrow in Rust.
+
+`pgeon <https://github.com/0x0L/pgeon>`_
+  A client that manually parses the Postgres wire format and produces
+  Arrow data, bypassing JDBC/ODBC. While it attempts to optimize this
+  case, the Postgres wire protocol is still row-oriented.
+
+`Turbodbc <https://turbodbc.readthedocs.io/en/latest/>`_
+  A set of Python ODBC bindings, implementing PEP 249, that also
+  provides APIs to fetch data as Arrow batches, optimizing the
+  conversion internally.
+
+Papers
+------
+
+Raasveldt, Mark, and Hannes Mühleisen. `“Don't Hold My Data Hostage -
+A Case for Client Protocol Redesign”`_. In *Proceedings of the VLDB
+Endowment*, 1022–1033, 2017.
+
+ADBC 1.0.0 Header
+=================
+
+.. literalinclude:: ../../../format/adbc.h
```

Comment:
Should we just reproduce the ADBC declarations here, instead of the full boilerplate?